tests/integration/helpers/cluster.py

```diff
@@ -1124,23 +1124,28 @@ def http_query_and_get_error(self, sql, data=None, params=None, user=None, passw
         return self.http_query(sql=sql, data=data, params=params, user=user, password=password,
                                expect_fail_and_get_error=True)
 
-    def stop_clickhouse(self, start_wait_sec=5, kill=False):
+    def stop_clickhouse(self, stop_wait_sec=30, kill=False):
         if not self.stay_alive:
             raise Exception("clickhouse can be stopped only with stay_alive=True instance")
 
         self.exec_in_container(["bash", "-c", "pkill {} clickhouse".format("-9" if kill else "")], user='root')
-        time.sleep(start_wait_sec)
+        deadline = time.time() + stop_wait_sec
+        while time.time() < deadline:
+            time.sleep(0.5)
+            if self.get_process_pid("clickhouse") is None:
+                break
+        assert self.get_process_pid("clickhouse") is None, "ClickHouse was not stopped"
 
-    def start_clickhouse(self, stop_wait_sec=5):
+    def start_clickhouse(self, start_wait_sec=30):
```

Comment:
Do we actually want to change default for all usages in tests?

Reply:
It doesn't make other tests worse, because in start/stop method we actually don't wait for the whole timeout checking every 0.5 seconds that CH is started/stopped. But it can help to slow tests under sanitizers without an explicit timeout setting.